Design/lazy plugin renderer

lcarva opened the pull request.

csomh commented on docs/build_parameters.rst:
```diff
@@ -254,10 +252,37 @@ to the root of the object::
+                          "description": "signing intent keys",
+                          "type": "array",
+                          "items": {
+                              "type": "string"
```
Could this be more restrictive? Like hexadecimal?

twaugh replied:
`"pattern": "^[a-fA-F0-9]+$"` ?

csomh commented on docs/build_parameters.rst:
```diff
@@ -254,10 +252,37 @@ to the root of the object::
+                "items": {
+                    "properties": {
+                      "name": {
+                        "description": "signing intent name",
```
Could this be made uniform, and start descriptions with a capital letter? (And not ending them with a period.)

lcarva commented:
Removed the schema completely out of design doc. It now references the one in atomic-reactor itself.
Added various changes to bring design doc up to speed.

lcarva commented:
Reviewers, ping.
